### PR TITLE
Verkkokauppa refactoring

### PR DESCRIPTION
**Pull request by ranta**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Refactor code related to upcoming Verkkokauppa changes.
  - Verkkokauppa changes will be made in two parts to make reviewing and resolving conflicts easier
- There should be little to no functional changes

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Automated tests

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3066

### CODE REVIEW

File: docker/uwsgi.yml [hunk 1]
```diff
@@ -45,7 +45,3 @@ uwsgi:
   ignore-sigpipe: true
   ignore-write-errors: true
   disable-write-exception: true
-
```
Comment (matti-lamppu) [inline docker/uwsgi.yml:48]: This monitoring is not related to Prometheus. It's for uWSGI, but I'm not completely sure what for. [Some statistics](https://dev.azure.com/City-of-Helsinki/tilavarauspalvelu/_git/tilavarauspalvelu-pipelines?path=/devops/tilavarauspalvelu-core-service.yml)?

Comment (matti-lamppu) [reply docker/uwsgi.yml:48]: I think we shouldn't remove it until we know what it does.

File: reservation_units/utils/reservation_unit_reservation_state_helper.py [hunk 2]
```diff
@@ -90,119 +120,132 @@ def __active_pricing_type_alias(cls) -> dict[str, Subquery]:
...
+        return (
+            # Pricing is paid and there is a payment product
+            Q(active_pricing_type=PricingType.PAID, payment_product__isnull=False)
+            # or pricing is free
+            | Q(active_pricing_type=PricingType.FREE)
         )
 
     @classmethod
     def __is_reservable(cls, reservation_unit: ReservationUnit) -> bool:
-        """
```
Comment (matti-lamppu) [inline reservation_units/utils/reservation_unit_reservation_state_helper.py:111]: Removed docstring here without adding a replacement.

File: reservation_units/utils/reservation_unit_reservation_state_helper.py [hunk 3]
```diff
@@ -90,119 +120,132 @@ def __active_pricing_type_alias(cls) -> dict[str, Subquery]:
...
     @classmethod
     def __get_is_reservable_query(cls) -> QueryState:
         return QueryState(
             aliases=cls.__active_pricing_type_alias(),
             filters=(cls.__is_reservable_period_query() & cls.__has_valid_pricing_query()),
         )
 
+    @classmethod
+    def __is_reservable_period(cls, now: datetime.datetime, reservation_unit: ReservationUnit) -> bool:
+        """
```
Comment (matti-lamppu) [inline reservation_units/utils/reservation_unit_reservation_state_helper.py:157]: Could add a written explanation like the ones above.

File: reservation_units/utils/reservation_unit_reservation_state_helper.py [hunk 4]
```diff
@@ -90,119 +120,132 @@ def __active_pricing_type_alias(cls) -> dict[str, Subquery]:
...
     def __get_is_scheduled_closing_query(cls) -> QueryState:
         return QueryState(
             aliases=cls.__active_pricing_type_alias(),
             filters=(cls.__is_scheduled_closing_period_query() & cls.__has_valid_pricing_query()),
         )
 
+    @classmethod
+    def __is_scheduled_closing_period(cls, now: datetime.datetime, reservation_unit: ReservationUnit) -> bool:
+        """
+        ┌────────┬────────┬────────┐
```
Comment (matti-lamppu) [inline reservation_units/utils/reservation_unit_reservation_state_helper.py:217]: Could add a written explanation like the ones above.

File: reservation_units/utils/reservation_unit_reservation_state_helper.py [hunk 5]
```diff
@@ -212,55 +255,58 @@ def __is_scheduled_closing_period_query(cls) -> Q:
     @classmethod
     def __is_reservation_closed(cls, reservation_unit: ReservationUnit) -> bool:
         """
-        Support following cases in regard to reservation_begins and reservation_ends:
+        ┌────────┬────────┬────────┐
```
Comment (matti-lamppu) [inline reservation_units/utils/reservation_unit_reservation_state_helper.py:258]: Could add a written explanation like the ones above.

**Review by matti-lamppu — CHANGES_REQUESTED** [review]
Should not remove the uWSGI monitoring port, otherwise OK.

File: docker/uwsgi.yml (hunk 1 above)
Comment (ranta) [reply docker/uwsgi.yml:48]: According to the [PR it was added in](https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/721), it should only be used for Prometheus-related stuffs 🤔

**Review by matti-lamppu — APPROVED** [review]

File: reservation_units/utils/reservation_unit_reservation_state_helper.py (hunk 3 above)
Comment (ranta) [reply reservation_units/utils/reservation_unit_reservation_state_helper.py:157]: Will add better documentation to all these when rewriting the helpers to be clearer a later date :)
